src/core/Akka/Actor/Stash/IWithStash.cs

```diff
@@ -30,7 +30,7 @@ namespace Akka.Actor
     /// that does not enforce any mailbox type see <see cref="IWithUnrestrictedStash"/>.
     /// </para>
     /// </summary>
-    public interface IWithStash : IWithUnrestrictedStash, IRequiresMessageQueue<IDequeBasedMessageQueueSemantics>
+    public interface IWithStash : IWithUnrestrictedStash, IRequiresMessageQueue<IBoundedDequeBasedMessageQueueSemantics>
     {
```

Comment:
Default deque must be unbounded, this is change in behavior. 

Keep in mind that when using these marker interfaces, the mailbox type will be resolved via the `akka.actor.mailbox.requirements` configuration in the hocon, not via the `StashFactory` (maybe that's the confusion).

Reply:
`IWithStash` is the replacement for `IWithBoundedStash` interface, the unbounded behaviour is `IWithUnboundedStash`

Reply:
Yeah this is an unsafe change.

Reply:
Oh wait, are these the new APIs @ismaelhamed added?

Reply:
Ah, I get it now... IWithStash is supposed to replace both `IWithBoundedStash` and `IWithUnboundedStash` and the behaviour is dictated by the settings?